pkg/k2/constructs/construct_evaluator_test.go

```diff
@@ -162,94 +187,140 @@ func TestInterpolateValue(t *testing.T) {
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: nil,
 		},
 		{
 			name:     "Non-existent resource",
 			rawValue: "${resources:nonExistentResource.prop}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			hasError: false,
 		},
 		{
 			name:     "Non-existent resource property",
 			rawValue: "${resources:testResource.nonExistentProp}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: nil,
 		},
 		{
 			name:     "Invalid array index",
 			rawValue: "${inputs:sliceInput[invalid]}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: nil,
 		},
 		{
 			name:     "Out of bounds array index",
 			rawValue: "${inputs:sliceInput[10]}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: nil,
 		},
 		{
 			name:     "Multiple interpolations in a string",
 			rawValue: "Start ${inputs:stringInput} middle ${inputs:intInput} end",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: "Start Hello middle 42 end",
 		},
 		{
 			name:     "Mixed string interpolation with slice interpolation",
 			rawValue: "${inputs:stringInput} ${inputs:sliceInput}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: "Hello [a b c]",
 		},
 		{
 			name:     "Mixed string interpolation with map interpolation",
 			rawValue: "${inputs:stringInput} ${inputs:mapInput}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			// the dynamic key has not been interpolated (that would occur in a separate step)
 			expected: "Hello map[${inputs:stringInput}:value3 key1:value1 key2:2]",
 		},
 		{
 			name:     "Mixed string interpolation with struct interpolation",
 			rawValue: "${inputs:stringInput} ${inputs:stringerStructInput}",
-			ctx:      NewInterpolationContext(mockConstruct, ResourceInterpolationContext),
+			data: DynamicValueData{
+				currentOwner: mockConstruct,
+			},
 			expected: "Hello Hello",
 		},
+		{
+			name:     "Go template interpolation",
+			rawValue: "Hello {{ .Inputs.stringInput }}",
+			data:     DynamicValueData{currentOwner: mockConstruct},
+			expected: "Hello Hello",
+		},
+		{
+			name:     "Go template evaluated before interpolation string",
+			rawValue: "Hello ${inputs:{{\"stringInput\"}}}",
```

Comment:
```suggestion
			rawValue: `Hello ${inputs:{{"stringInput"}}}`,
```
nit: a little nicer to use backticks to avoid needing escapes